examples/addsvc/client/httpjson/factory.go

```diff
@@ -6,45 +6,58 @@ import (
 
 	"github.com/go-kit/kit/endpoint"
 	"github.com/go-kit/kit/examples/addsvc/server"
+	"github.com/go-kit/kit/loadbalancer"
+	kitot "github.com/go-kit/kit/tracing/opentracing"
 	httptransport "github.com/go-kit/kit/transport/http"
+	"github.com/opentracing/opentracing-go"
 )
 
-// SumEndpointFactory transforms a http url into an Endpoint.
+// SumEndpointFactory generates a Factory that transforms an http url into an
+// Endpoint.
+//
 // The path of the url is reset to /sum.
-func SumEndpointFactory(instance string) (endpoint.Endpoint, io.Closer, error) {
-	sumURL, err := url.Parse(instance)
-	if err != nil {
-		return nil, nil, err
+func NewSumEndpointFactory(tracer opentracing.Tracer) loadbalancer.Factory {
```

Comment:
I guess it would be MakeSumEndpointFactory rather than New, and there's some discontinuity between the function name and the comment, but I'll leave this class of review off until the golden master, if that's OK.